### ISSUE TITLE
Support for the new "Energy HA integration"

### ISSUE THREAD

**Issue by pluskal** [weltenwort/home-assistant-rct-power-integration#91]
**Is your feature request related to a problem? Please describe.**
Home Assistant announced a new way to monitor Energy. In order to integrate with that, appropriate sensors need to be configured for [Long-term Statistics](https://developers.home-assistant.io/docs/core/entity/sensor/#long-term-statistics).

**Describe the solution you'd like**
Set sensor `state_class` to `measurement` for sensors monitoring current `production, consumption, grid, battery, etc.`

**Describe alternatives you've considered**
None

**Additional context**
[Home Assistant 2021.8 Announcement for Home Energy Management](https://www.home-assistant.io/blog/2021/08/04/home-energy-management/)

**Comment by weltenwort** [comment]
@pluskal thanks for filing this issue! The energy integration looks exciting and I'll investigate supporting it as soon as I find the time.

**Comment by weltenwort** [comment]
I tried to implement this in #98 but it takes a while for the statistics to become available. I'll keep an eye on it over the next days.

If you are a user of this integration (or a developer of a similar one) I'd be interested in hearing whether it works for you.

**Comment by weltenwort** [comment]
Looks good so far:

![image](https://user-images.githubusercontent.com/973741/128608901-737c8679-10ec-4bd6-89a0-910bb0443ba2.png)

Apparently one has to wait at least a few hours for values to show up.

**Comment by pluskal** [comment]
Nice! You made it very, very fast :). Thanks!

It seems to work flawlessly.

![image](https://user-images.githubusercontent.com/3617487/128628075-ab755d3f-7874-4b8f-8180-bb72bc6a0139.png)

**Comment by weltenwort** [comment]
Something might not be right with the grid production. It stays fixed a 0 kWh even though it's feeding to the grid. Unfortunately there is no documentation about whether the different numbers are expected to be positive or negative. I'll have to dig into the HA core code to find that out. :eyes:

**Comment by pluskal** [comment]
I think that It is ok in my case. I have disabled `grid production` in the RCT application for now.

**Comment by weltenwort** [comment]
There seem to be quite a lot of confusion about the way the energy dashboard works. I'll wait until it has settled a bit and until more built-in integrations support it so I can learn from them. Either that or until a proper documentation exists for how energy sensors interact with the energy dashboard.

I'll reopen the issue in the meantime in order to be transparent about the limitations.

**Comment by k3mpaxl** [comment]
The reason why grid production is not working well is that its a negative value. I solved this issue with these few lines:

```
template:
  - sensor:
      - name: "RCT Grid Energy Production Day Inverted Value"
        unit_of_measurement: "Wh"
        state_class: "measurement"
        device_class: "energy"
        state: >
          {% set production = states('sensor.rct_power_storage_grid_energy_production_day') | float %}

          {{ (production * -1) |round(1) }}
        attributes:
          last_reset: "{{ state_attr('sensor.rct_power_storage_grid_energy_production_day', 'last_reset') }}"

```

<img width="609" alt="rct" src="https://user-images.githubusercontent.com/2837689/130360592-60a73026-e0a1-4c00-8b43-0d93ee1c2595.png">

**Comment by weltenwort** [comment]
@k3mpaxl thanks for sharing that. It's what I almost suspected. Maybe I should add a positive alternative for the grid feed sensor.

**Comment by hanzsez** [comment]
is there something extra which has to be added/manualy modified to make this intergration work with the Energy panel? I installed clean home assistant in docker, manually added the integration, added the invertor with bateries so they correctly show all the possible values. But even after several hours when I try to enroll to the energy meter data, it tells me there are no data source listed to pick the statistics from.

**Comment by k3mpaxl** [comment]
Home assistant changed from the state requirements from measurement to total_increasing in version 2021.09.02.

Since a few days the integration got updated and there are total entities that can be selected:

![Screenshot_20210908_082717_io.homeassistant.companion.android_edit_312152347035700.jpg](https://user-images.githubusercontent.com/2837689/132457847-76bfa742-db38-4808-bf81-34592ae331f7.jpg)

**Comment by weltenwort** [comment]
Yes, the fact that the inverter reports the produced energy with a negative sign is inconvenient. The refactorings I did as part of #120 will make it easier to add "calculated" entities. I'm planning to expose an entity for the absolute value of the produced energy with the correct sign as well as the sum of the energy produced by the two generators. Hope I'll get around to it this weekend.

**Comment by weltenwort** [comment]
I just released [v0.8.1](https://github.com/weltenwort/home-assistant-rct-power-integration/releases/tag/v0.8.1), which includes a new entity `"Grid Energy Production Absolute Total"` that has the sign the energy dashboard expects. I left the original entity in place for backwards-compatibility.
